tests/test_keras_integration.py

```diff
@@ -178,12 +179,56 @@ def test_save_pretrained(self):
 
         model.build((None, 2))
 
-        save_pretrained_keras(model, f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        save_pretrained_keras(
+            model,
+            f"{WORKING_REPO_DIR}/{REPO_NAME}",
+        )
         files = os.listdir(f"{WORKING_REPO_DIR}/{REPO_NAME}")
 
         self.assertIn("saved_model.pb", files)
         self.assertIn("keras_metadata.pb", files)
         self.assertEqual(len(files), 4)
+        model = from_pretrained_keras(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        self.assertIsNone(model.optimizer)
+
+    def test_save_pretrained_optimizer_state(self):
+        REPO_NAME = repo_name("save")
+        model = self.model_init()
+
+        model.build((None, 2))
+
+        save_pretrained_keras(
+            model, f"{WORKING_REPO_DIR}/{REPO_NAME}", include_optimizer=True
+        )
+        files = os.listdir(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+
+        self.assertIn("saved_model.pb", files)
+        self.assertIn("keras_metadata.pb", files)
+
+        self.assertEqual(len(files), 4)
+        model = from_pretrained_keras(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        self.assertIsNotNone(model.optimizer)
+
+    def test_save_pretrained_kwargs(self):
+        REPO_NAME = repo_name("save")
+        model = self.model_init()
+
+        model.build((None, 2))
+
+        save_pretrained_keras(
+            model,
+            f"{WORKING_REPO_DIR}/{REPO_NAME}",
+            include_optimizer=False,
+            save_traces=False,
+        )
+        files = os.listdir(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+
+        self.assertIn("saved_model.pb", files)
+        self.assertIn("keras_metadata.pb", files)
+        self.assertEqual(len(files), 4)
```

Comment:
Feel free to remove things if they are well tested in other tests

Reply:
this is already tested in the test below but not the DummyModel's weights so I'll remove this part only